highfidelity/hifi
# Migrate to single QML engine

This PR attempts to improve the stability of the QML & Web rendering systems.

* Use a single QML engine for all QML, rather than an engine per surface.  
* Each surface will now have a specific [QQmlContext](http://doc.qt.io/qt-5/qqmlcontext.html) used to store surface-specific properties such as the surface dimensions, the event bridge object, etc.
* The engine root QQmlContext will be used to store unchanging values that are applicable to all QML surfaces, such as the JavaScript event bridge initialization code, the pointer to the application, etc.

Additionally, this PR moves all of the custom types derived from `QQuickWebEngineProfile` out of the Interface project and into the Networking project.  Our current QML code instantiates multiple instances of these, often pointing to the same underlying filesystem location for storage.  It's unclear whether this usage is supported.  Because all QML surfaces now share a single engine and a root QML context, these can now be exposed to QML by name rather than having the QML instantiate them, so there is only one profile instance for each backing storage location. 

## Testing

* Ensure desktop UI still works
* Ensure script managed UI such as the tablet, and web or QML based overlays still work
* Ensure that WebEntity rendering still works
